python/pyarrow/_dataset.pyx

```diff
@@ -245,9 +245,10 @@ cdef class Dataset(_Weakrefable):
             The columns will be passed down to Datasets and corresponding data
             fragments to avoid loading, copying, and deserializing columns
             that will not be required further down the compute chain.
-            By default all of the available columns are projected. Raises
-            an exception if any of the referenced column names does not exist
-            in the dataset's Schema.
+            By default all of the available columns along with the augmented fields
+            such as `batch_index`, `fragment_index`, `last_in_fragment` and `filename`
+            are projected. Raises an exception if any of the referenced column names 
```

Comment:
This seems to contradict the test above, isn't it checking that these fields don't exist by default?

Reply:
Yes, sorry I missed that, changed the doc now, can you check again please.